flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/CalcRankTransposeRule.scala

```diff
@@ -166,7 +166,7 @@ class CalcRankTransposeRule
     val oldOrderKey = rank.orderKey
     val oldFieldCollations = oldOrderKey.getFieldCollations
     val newFieldCollations = oldFieldCollations.map {
-      fc => fc.copy(fieldMapping(fc.getFieldIndex))
+      fc => fc.withFieldIndex(fieldMapping(fc.getFieldIndex))
```

Comment:
deprecated at https://github.com/apache/calcite/commit/513f4d2510a47e8d0b55f35dac56785c16fa7d5f
in favor of `withXXX` methods